app/models/teacher_profile.rb

```diff
@@ -9,7 +9,9 @@ class TeacherProfile < ApplicationRecord
   # TODO: Legacy associations, to be removed
   has_one :early_career_teacher_profile, -> { active_record }, class_name: "ParticipantProfile::ECT"
   has_one :mentor_profile, -> { active_record }, class_name: "ParticipantProfile::Mentor"
-  has_one :ecf_profile, -> { active_record }, class_name: "ParticipantProfile::ECF"
+
+  has_many :ecf_profiles, -> { active_record.includes(:cohort).order("cohort.start_year DESC") }, class_name: "ParticipantProfile::ECF"
```

Comment:
I hope the answer is no, but does the fact that this table has a cohort column confuse matters?

Reply:
I think it could. I made it more explicit, so hopefully not anymore.